File: serveur/app/models/recipe.py
# app/models/recipe.py
#
# Nutrition V1 (2026-06-12, cf. docs/NUTRITION_DESIGN.md §3.3).
# Plats composes ET repas enregistres (D7 : une seule entite couvre les deux).
# Type A user-scoped (cascade ownership politique 8).
# kind UPPER_CASE (politique 11) :
#   - RECIPE     : plat (macros au prorata du poids consomme, total_weight_g gere le ratio cru/cuit)
#   - SAVED_MEAL : repas enregistre (insertion des ingredients tels quels)

from sqlalchemy import Column, Integer, String, Float, ForeignKey, DateTime
from app.database import Base
import uuid


class Recipe(Base):
    __tablename__ = "recipes"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    uuid = Column(String, nullable=False, unique=True, default=lambda: str(uuid.uuid4()))

    name = Column(String, nullable=False)              # user-typed, non traduit
    kind = Column(String, nullable=False)              # RECIPE | SAVED_MEAL (pas de default : choix explicite)
    total_weight_g = Column(Float, nullable=True)      # poids final cuit (kind=RECIPE seulement)

    updated_at = Column(DateTime(timezone=True), nullable=True)

    def __repr__(self):
        return (
            f"Recipe(id={self.id}, user_id={self.user_id}, uuid='{self.uuid}', "
            f"name='{self.name}', kind='{self.kind}')"
        )

File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/Recipe.kt
package com.example.sportapp.core.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601

/**
 * Nutrition A1 (2026-06-17) — portage Android du modèle serveur `recipes`
 * (cf. serveur/app/models/recipe.py). Plats composés ET repas enregistrés
 * (D7 : une seule entité couvre les deux). Type A user-scoped.
 *
 * `kind` UPPER_CASE (politique 11) :
 *   - RECIPE     : plat (macros au prorata du poids consommé via total_weight_g)
 *   - SAVED_MEAL : repas enregistré (insertion des ingrédients tels quels)
 */
@Entity(
    tableName = "recipes",
    indices = [
        Index(value = ["uuid"], unique = true),
        Index(value = ["user_id"]),
    ]
)
data class Recipe(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "user_id") val userId: Int,

    val name: String,                                                  // user-typed, non traduit
    val kind: String,                                                  // RECIPE | SAVED_MEAL
    @ColumnInfo(name = "total_weight_g") val totalWeightG: Float? = null,   // kind=RECIPE seulement

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)
